### ISSUE TITLE
Support for API v2.0

### ISSUE THREAD

**Issue by jjatria** [LibreCat/WWW-ORCID#2]
On February 14th, ORCID released [a new version of their API](https://orcid.org/content/register-client-application).

According [to them](https://orcid.org/blog/2017/02/14/all-about-our-new-api-interview-rob-peters-director-technology):

> We are aiming to sunset 1.2 in late 2017. Regardless of the sunset date, if you agree with ORCID’s mission and care about researchers interact with ORCID you’ll want to move to 2.0 now.

What is the plan for changes in this distribution to support 2.0? Since the old API will sunset as early as late this year, I imagine having a new namespace (such as `WWW::ORCID::v2` or something like that) is not so useful. 

I haven't looked at this in detail, but I'm thinking it could be specified in the constructor. As in

```perl
# From now and for some time
# Maybe issuing a warning?
my $old = WWW::ORCID->new;
my $new = WWW::ORCID->new(
  version => 2.0,
);

# From that cutoff time until "late 2017"
my $old = WWW::ORCID->new(
  version => 1.0,
);
my $new = WWW::ORCID->new;
``` 

That way, we ensure not to break compatibility too suddenly, and we maintain the current namespace.

If this looks reasonable, I can submit a PR.

**Comment by vpeil** [comment]
I'm also interested in an update for version 2!

**Comment by nics** [comment]
We're definitely planning on supporting v2. Your proposal is in line with what i was planning to do, so feel free to PR! If we want a clean separation of the code we can still use the version parameter to delegate to a v1 or v2 backend package.

**Comment by nics** [comment]
There is now a developer release available that supports most of the 2.0 public and member api's: http://search.cpan.org/~nics/WWW-ORCID-0.03_01/. Support for 2.1 will also be added, the client backends are pluggable now.

You can install the dev release with `cpanm --dev WWW::ORCID` or `cpanm NICS/WWW-ORCID-0.03_01.tar.gz`. Any feedback will be appreciated.

I decided to remove support for 1.0 though, it's end of life is not far off anymore and the old client was incomplete anyway. To maintain compatibility when the new version is released, lock your version to 0.02.

**Comment by nics** [comment]
Version 0.0401 supports the 2.0 api.
